### DISCUSSION TITLE
v2.3.4 AqualinkD Setup & MQTT Questions

### DISCUSSION THREAD

**Discussion by jmp2000** [aqualinkd/AqualinkD#292] [Q&A] [answered]
Good afternoon!  Making good progress on the setup.   Have 3 questions, 2 getting AqualinkD settings tweaked, and then 1 getting everything working in Home Assistant with MQTT.  Right now most things works well off the WebUI settings, but want to get fully integrated into Home Assistant ultimately.

Install details:  Raspberry Pi Zero 2 W for AqualinkD (v2.3.4 latest); RPi4 running as Home Assistant for the house

2 attachments:  Aqualink.conf file from the Pi Zero 2 W, and the Home Assistant configuration.yaml file from the Pi4

1. One last AqualinkD control point needed - All switches, lights, etc. are working for the pool except this yellow highlighted "water feature".  It controls a Jandy servo to send pool pump water down a pipe to some deck jets in the shallow end.   I used the serial logging feature to verify that this was a "RS-2/6 Dual" board (QQ Firmware).  Tried every button and everything else is working great through AqualinkD, but cannot figure out how to control that last servo valve.   Works fine from the control panel outside or from the Jandy Aqualink RS OneTouch.  (Photo and AqualinkD.conf attached below).

2. Is there a way to rename the "Pool Air Temperature" in AqualinkD WebUI to something else?  Say I wanted to call it "Air Temperature", is that possible to reconfigure in the AqualinkD.conf file somewhere?  Or another way?

3. MQTT setup is stumping me.  I've got the Mosquitto / MQTT Broker configured in Home Assistant on my RPi4.   When I run AqualinkD on the PiZero, it doesn't seem to connect?   The PiZero has the AqualinkD pointing to port 1883, and with the Home Assistant MQTT login and password.   The RPi4 running home assistant, has the MQTT Mosquitto Broker running.   HomeAssistant shows there is a new connection and then says the unknown client closed its connection.  I'm sure there's something simple that me as a noob is totally missing =)  Appreciate any kind coaching to get that MQTT connection going between these 2 Pi's.

Thanks so much - really appreciate the community here!  (And what a fantastic product in AqualinkD!!)

Here's what's in the AqualinkD logs:
Notice:NetService:Starting MQTT client to myAqualinkDIPaddress:1883
Warning:NetService:MQTT Connection closed
Notice:NetService:Starting MQTT client to myAqualinkDIPaddress:1883
Warning:NetService:MQTT Connection closed

Here's what the Home Assistant Mosquitto Broker logs say:
2024-05-11 15:29:03: New connection from adifferentIPaddress:5digits on port 1883.
2024-05-11 15:29:03: Client <unknown> closed its connection.
2024-05-11 15:31:03: New connection from adifferentIPaddress:5digits on port 1883.
2024-05-11 15:31:03: Client <unknown> closed its connection.

![Pool control panel](https://github.com/sfeakes/AqualinkD/assets/149847986/49fe651e-6f3e-4d72-be89-efc0fa8d339f)

[aqualinkd 05-11-2024.txt](https://github.com/sfeakes/AqualinkD/files/15284156/aqualinkd.05-11-2024.txt)

[HomeAssistant configuration-yaml.txt](https://github.com/sfeakes/AqualinkD/files/15284220/HomeAssistant.configuration-yaml.txt)

**Comment by sfeakes** [comment]
#1. Looks like your install has used the solar-heater as the water feature.  On a 2/6 Dual panel, that's button 11.  So in your config you'll need something like below.
```
button_11_label=Water Feature
```

Also by default, solar_heater is hidden from the web UI (since very few use it). So you need to unhide it.
In file `/var/www/aqualinkd/config.js` around line 40 change
The below
```
        "Spa_Water",
        "Freeze_Protect",
        "CHEM/pH",
        "CHEM/ORP"
        //"Solar_Heater",
      ];
```
To This. (**_remove // and ADD the , after "CHEM/ORP"_**)
```
"Spa_Water",
        "Freeze_Protect",
        "CHEM/pH",
        "CHEM/ORP",
        "Solar_Heater",
      ];
```

#2 Sorry, that's fixed.  Most home hubs have air temp and water temp, so I added pool to be specific.  I'll look at making this a configuration option.

#3 You will probably have to put the MQTT Broker on Home Assistant in debug mode, and get some more details on why it's dropping the connection.

**Reply by jmp2000** [reply]
Perfect - that fixed up item 1 & 2.  I will keep learning on MQTT.

**Reply by sfeakes** [reply, marked as answer]
For testing MQTT, it's probably easier to install mosquitto_sub on the Pi running AqualinkD, then run the below to test the connection.

```mosquitto_sub -v -h <mosquitto_ip_address> -p 1883 -t '#' -u <user> -P <password>```

If that connects without errors and starts printing messages, then let me know. my guess is it will fail and give you some more information as to why.

mosquitto_sub is part of mosquitto-clients tools. So the below will install that for you.
```sudo apt install mosquitto-clients```

**Reply by jmp2000** [reply]
Thanks!  I think the combination of installing the Mosquitto clients on the Pi Zero running AqualinkD and also ensuring I got the IP address corrected to the HA machine in the AqualinkD config file has it connected now!   Appreciate the patience while I'm learning!
